### PR TITLE
Add support for `zpopmin`

### PR DESCRIPTION
**Pull request by eliangcs**
Matches the `zpopmin(key, count=None)` function signature in [redis-py](https://redis-py.readthedocs.io/en/stable/#redis.Redis.zpopmin). I could add `zpopmax` while I was there but that's all we need for now.

### CODE REVIEW

File: fakeredis/_server.py [hunk 1]
```diff
@@ -2129,6 +2129,20 @@ def zincrby(self, key, increment, member):
     def zlexcount(self, key, min, max):
         return key.value.zlexcount(min.value, min.exclusive, max.value, max.exclusive)
 
+    @command((Key(ZSet),), (Int,))
+    def zpopmin(self, key, count=None):
+        zset = key.value
+        min_score = ScoreTest.decode(b'-inf').lower_bound
+        max_score = ScoreTest.decode(b'inf').upper_bound
+        items = list(zset.irange_score(min_score, max_score))
```
Comment (bmerry) [inline fakeredis/_server.py:2137]: I don't think it's necessary to construct and pass `min_score` and `max_score`: you should be able to just pass `None` for both.

File: fakeredis/_server.py [hunk 2]
```diff
@@ -2129,6 +2129,20 @@ def zincrby(self, key, increment, member):
     def zlexcount(self, key, min, max):
         return key.value.zlexcount(min.value, min.exclusive, max.value, max.exclusive)
 
+    @command((Key(ZSet),), (Int,))
+    def zpopmin(self, key, count=None):
```
Comment (bmerry) [inline fakeredis/_server.py:2133]: Unfortunately the `@command` decorator doesn't support optional (0 or 1 instance) arguments yet, only repeating arguments (0 or more instances), so you need to make the argument a `*args` and then check in the function if there are too many arguments (see `_zrangebylex` for an example).

This will also need a unit test, which can be done with `r.execute_command` to directly pass parameters.

File: test/test_fakeredis.py [hunk 3]
```diff
@@ -2112,6 +2112,37 @@ def test_zadd_incr(r, ch):
...
+def test_zpopmin_empty(r):
+    assert r.zpopmin('foo') == []
+
+
+def test_zpopmin_default_count(r):
+    zadd(r, 'foo', {'one_a': 1})
+    zadd(r, 'foo', {'one_b': 1})
+    zadd(r, 'foo', {'two_a': 2})
+    assert r.zpopmin('foo') == [(b'one_a', 1.0)]
+    assert r.zcard('foo') == 2
```
Comment (bmerry) [inline test/test_fakeredis.py:2124]: Rather test the actual contents (with `r.zrange(0, -1)`) to ensure that the right elements were removed.

File: test/test_fakeredis.py [hunk 4]
```diff
@@ -2112,6 +2112,37 @@ def test_zadd_incr(r, ch):
...
 
 
+def test_zpopmin_empty(r):
+    assert r.zpopmin('foo') == []
+
+
+def test_zpopmin_default_count(r):
+    zadd(r, 'foo', {'one_a': 1})
+    zadd(r, 'foo', {'one_b': 1})
+    zadd(r, 'foo', {'two_a': 2})
```
Comment (bmerry) [inline test/test_fakeredis.py:2122]: Could do all three steps in one go, but more importantly, it would be better to use data where the elements with the lowest scores are not the ones that are first lexicographically, to ensure that it is really scores being considered.

File: test/test_fakeredis.py [hunk 5]
```diff
@@ -2112,6 +2112,37 @@ def test_zadd_incr(r, ch):
...
+    assert r.zpopmin('foo') == [(b'one_a', 1.0)]
+    assert r.zcard('foo') == 2
+
+
+def test_zpopmin_zero_count(r):
+    zadd(r, 'foo', {'one_a': 1})
+    zadd(r, 'foo', {'one_b': 1})
+    zadd(r, 'foo', {'two_a': 2})
+    assert r.zpopmin('foo', count=0) == []
+    assert r.zcard('foo') == 3
```
Comment (bmerry) [inline test/test_fakeredis.py:2132]: Again, test the contents.

File: test/test_fakeredis.py [hunk 6]
```diff
@@ -2112,6 +2112,37 @@ def test_zadd_incr(r, ch):
...
+    zadd(r, 'foo', {'two_a': 2})
+    assert r.zpopmin('foo') == [(b'one_a', 1.0)]
+    assert r.zcard('foo') == 2
+
+
+def test_zpopmin_zero_count(r):
+    zadd(r, 'foo', {'one_a': 1})
+    zadd(r, 'foo', {'one_b': 1})
+    zadd(r, 'foo', {'two_a': 2})
+    assert r.zpopmin('foo', count=0) == []
```
Comment (bmerry) [inline test/test_fakeredis.py:2131]: Can you also test with `-1` (`pytest.mark.parametrize` is good for this).

File: test/test_fakeredis.py [hunk 7]
```diff
@@ -2112,6 +2112,37 @@ def test_zadd_incr(r, ch):
     assert zadd(r, 'foo', {'three': 1.0}, incr=True, xx=True, ch=ch) == 4.0
 
 
+def test_zpopmin_empty(r):
+    assert r.zpopmin('foo') == []
```
Comment (bmerry) [inline test/test_fakeredis.py:2116]: Can you also add a test that doesn't specify a count but for which the set is not empty, to test the code that defaults it to 1.

File: test/test_fakeredis.py [hunk 8]
```diff
@@ -2112,6 +2112,37 @@ def test_zadd_incr(r, ch):
     assert zadd(r, 'foo', {'three': 1.0}, incr=True, xx=True, ch=ch) == 4.0
 
 
+def test_zpopmin_empty(r):
```
Comment (bmerry) [inline test/test_fakeredis.py:2115]: It looks like all the unit tests will need a `@redis3_only` decorator since redis-py 2.x didn't implement the command.

**Comment by bmerry** [comment]
Fakeredis is under new maintenance at https://github.com/dsoftwareinc/fakeredis-py. I am thus closing all PRs on this repository. If you feel your PR is still relevant, please open a new one on that repository.
